File: src/main/level2/탑/Solution.java
package main.level2.탑;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Solution {
    public int[] solution(int[] heights) {
        int[] answer = new int[heights.length];

        List<Integer> answerList = new ArrayList<Integer>();
        boolean isBottom = true;
        for (int i = heights.length - 1; i >= 0; i--) {
            for (int j = i - 1; j >= 0; j--) {
                if (heights[j] > heights[i]) {
                    answerList.add(j + 1);
                    isBottom = false;
                    break;
                }
            }
            if (isBottom) {
                answerList.add(0);
            }
            isBottom = true;
        }

        Collections.reverse(answerList);
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }

        return answer;
    }
}

File: src/main/level2/탑/solution.py
# 다른 사람의 풀이를 보고 개선한 코드
def solution(heights):
    answer = [0] * len(heights)

    for i in range(len(heights) - 1, 0, -1):
        for j in range(i - 1, -1, -1):
            if heights[i] < heights[j]:
                answer[i] = j + 1
                break

    return answer